### ISSUE TITLE
spaceship - developerservices2 - 'Your session has expired. Please log in.'

### ISSUE THREAD

**Issue by fastlanebot** [fastlane/fastlane#2054]
<table>
      <tr>
        <td>
          <img src='https://avatars0.githubusercontent.com/u/2692482?v=3&s=70' width='35'>
        </td>
        <td>
          Original issue by @egarlock - Imported from <a href='https://github.com/fastlane/spaceship/issues/135'>fastlane/spaceship#135</a>
        </td>
      </tr>
    </table>

    [14:53:12]: Starting login with user 'MyAppleId@domain.com'
    /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/client.rb:255:in `parse_response': {"responseId"=>"2e9ec05d-8704-4363-bb0d-0f6c3e97c78d", "resultCode"=>1100, "resultString"=>"authentication.failed", "userString"=>"Your session has expired.  Please log in.", "creationTimestamp"=>"2015-10-22T19:53:14Z", "protocolVersion"=>"QH65B2", "requestId"=>nil, "userLocale"=>"en_US", "requestUrl"=>"https://developerservices2.apple.com/services/QH65B2/listTeams.action", "httpCode"=>200} (Spaceship::Client::UnexpectedResponse)
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/portal/portal_client.rb:58:in `teams'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/portal/ui/select_team.rb:35:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/portal/portal_client.rb:74:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/portal/spaceship.rb:37:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.12.2/lib/spaceship/portal/spaceship.rb:81:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/cert-1.0.0/lib/cert/runner.rb:15:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/cert-1.0.0/lib/cert/runner.rb:4:in `launch'
	from /Library/Ruby/Gems/2.0.0/gems/cert-1.0.0/bin/cert:32:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:68:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/cert-1.0.0/bin/cert:38:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/cert-1.0.0/bin/cert:44:in `<top (required)>'
	from /usr/local/bin/cert:23:in `load'
	from /usr/local/bin/cert:23:in `<main>'
Build step 'Execute shell' marked build as failure
Finished: FAILURE

**Comment by bgerstle** [comment]
this may be a dupe of #1996

**Comment by ricka-github** [comment]
I just started to use match and I'm also getting this error when trying to use a dedicated account for match. It works fine if I use my main account (which I've been using before I started with match).

**Comment by bgerstle** [comment]
@ricka-github does the dedicated account you're using also have an iTunesConnect user associated with it?

**Comment by ricka-github** [comment]
No it doesn't. They are only Apple developer accounts. I tried once with an admin account and once as a normal account, and I get the error on both counts.

Here is the output btw:
`+-----------------------+--------------------------------------------------------------------------+
|                                     Summary for match 0.4.0                                      |
+-----------------------+--------------------------------------------------------------------------+
| verbose               | true                                                                     |
| git_url               | https://mygitrepo... |
| type                  | development                                                              |
| app_identifier        | net.mydomain.*                                               |
| keychain_name         | login.keychain                                                           |
| readonly              | false                                                                    |
| force                 | false                                                                    |
| shallow_clone         | false                                                                    |
| force_for_new_devices | false                                                                    |
| skip_docs             | false                                                                    |
+-----------------------+--------------------------------------------------------------------------+

INFO [2016-04-11 19:20:54.61]: Cloning remote git repo...
INFO [2016-04-11 19:20:54.61]: $ git clone 'https://mygitrepo' '/var/folders/_5/txbsf9ds3q1ccy2fcsxvlsmh0000gn/T/d20160411-469-1j6shp6'
INFO [2016-04-11 19:20:55.04]: ▸ Cloning into '/var/folders/_5/txbsf9ds3q1ccy2fcsxvlsmh0000gn/T/d20160411-469-1j6shp6'...
INFO [2016-04-11 19:20:56.54]: ▸ warning: You appear to have cloned an empty repository.
INFO [2016-04-11 19:20:56.54]: ▸ Checking connectivity... done.
INFO [2016-04-11 19:20:56.55]: 🔓  Successfully decrypted certificates repo
WARN [2016-04-11 19:20:56.55]: To not be asked about this value, you can specify it using 'username'
Your Apple ID Username: mysharedappleid
INFO [2016-04-11 19:21:04.74]: Verifying that the certificate and profile are still valid on the Dev Portal...
INFO [2016-04-11 19:21:11.27]: 🔒  Successfully encrypted certificates repo
/Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/client.rb:427:in `parse_response': [!] Apple provided the following error info: (Spaceship::Client::UnexpectedResponse)
	authentication.failed
	Your session has expired.  Please log in.
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/portal/portal_client.rb:51:in `teams'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/portal/ui/select_team.rb:35:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/portal/portal_client.rb:67:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/portal/spaceship.rb:37:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.25.1/lib/spaceship/portal/spaceship.rb:81:in `select_team'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/lib/match/spaceship_ensure.rb:20:in `initialize'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/lib/match/runner.rb:11:in `new'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/lib/match/runner.rb:11:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/bin/match:52:in `block (3 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/bin/match:108:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/match-0.4.0/bin/match:116:in `<top (required)>'
	from /usr/local/bin/match:23:in `load'
	from /usr/local/bin/match:23:in `<main>'`

**Comment by TKBurner** [comment]
@ricka-github Thanks for following up. We are going to look into this issue. In the meantime, can you let me know if you are getting it consistently or intermittently? Thanks!

**Comment by ricka-github** [comment]
I am getting this consistently.

**Comment by peywen** [comment]
I got this error too. Clearing credential from cache didn't work either.

> Starting login with user 'ttdevios@gmail.com'
[!] Apple provided the following error info:
	authentication.failed
	Your session has expired.  Please log in.

from spaceship log:
[16:54:06]: >> GET: https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa 
[16:54:07]: << GET: https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa: 
[16:54:07]: >> POST: https://developerservices2.apple.com/services/QH65B2/listTeams.action 
[16:54:07]: << POST: https://developerservices2.apple.com/services/QH65B2/listTeams.action: {"responseId"=>"4bcfde6a-94e5-47d6-875e-1d8a3a4b2649", "resultCode"=>1100, "resultString"=>"authentication.failed", "userString"=>"Your session has expired.  Please log in.", "creationTimestamp"=>"2016-04-12T08:54:07Z", "protocolVersion"=>"QH65B2", "userLocale"=>"en_US", "requestUrl"=>"https://developerservices2.apple.com/services/QH65B2/listTeams.action", "httpCode"=>200}

**Comment by i2amsam** [comment]
@ricka-github @peywen could you two logging as described in this comment?
https://github.com/fastlane/fastlane/issues/4157#issuecomment-208566618

**Comment by ricka-github** [comment]
Simply logging in to appleid.apple.com as described in that comment fixed it for me. Thanks.

**Comment by TKBurner** [comment]
Awesome @ricka-github! Thanks for letting us know. I'm going to close this issue out, but if anyone is still affected after trying to log in to https://appleid.apple.com/ once with the email address they are using let us know. 👍

**Comment by peywen** [comment]
Awesome! This really fixed my issue, too. Thanks.

**Comment by mdamjanic7** [comment]
Hey guys, I'm still having this problem. I created a new Apple ID just for match, and I constantly keep having this issue. Going to [appleid.apple.com](appleid.apple.com) doesn't show anything out of the ordinary (username is an email address, no missing security questions...). 
I also tried using my personal account (again, no problems when logging in through [appleid.apple.com](appleid.apple.com)), but I keep getting the same message.

The only thing that might be out of the ordinary is that it is an Enterprise account. I checked #1790, #4157 and #4172 but I still have no idea what to do.
